v2/dtos/common/metrics.go

```diff
@@ -18,3 +23,26 @@ type MetricsResponse struct {
 	MemTotalAlloc  uint64 `json:"memTotalAlloc"`
 	CpuBusyAvg     uint8  `json:"cpuBusyAvg"`
 }
+
+// NewMetricsResponse creates new MetricsResponse with all fields set appropriately
+func NewMetricsResponse(
```

Comment:
We had better create another Metrics struct for this, because the client code might look like.
m := NewMetricsResponse(1000, 1234, 5647, 2353, 36554, 2654, 56)

It is easy to fill out the wrong number and need to change the function signature if we add more metrics.

```
type Metrics struct {
	MemAlloc       uint64 `json:"memAlloc"`
	MemFrees       uint64 `json:"memFrees"`
	MemLiveObjects uint64 `json:"memLiveObjects"`
	MemMallocs     uint64 `json:"memMallocs"`
	MemSys         uint64 `json:"memSys"`
	MemTotalAlloc  uint64 `json:"memTotalAlloc"`
	CpuBusyAvg     uint8  `json:"cpuBusyAvg"`
}

type MetricsResponse struct {
	Versionable    `json:",inline"`
	Metrics    `json:",inline"`
}

func NewMetricsResponse(m Metrics) MetricsResponse {
...
```

Then, users could call the factory method like this

`m := NewMetricsResponse(Metrics{MemAlloc: 1024, MemFrees: 1000, ....})`

Reply:
Yea, I was on the fence about creating the struct, so your feedback has pushed me over it.. ;-) Will create the struct.

Reply:
Changed. Much cleaner! Thanks for feedback!